homeassistant/components/climate/__init__.py

```diff
@@ -345,56 +349,58 @@ def _report_turn_on_off(feature: str, method: str) -> None:
             else:
                 message = (
                     "Entity %s (%s) implements HVACMode(s): %s and therefore implicitly"
-                    " supports the %s service without setting the proper"
+                    " supports the %s methods without setting the proper"
                     " ClimateEntityFeature. Please %s"
                 )
             _LOGGER.warning(
                 message,
                 self.entity_id,
                 type(self),
                 feature,
-                feature.lower(),
+                method,
                 report_issue,
             )
 
         # Adds ClimateEntityFeature.TURN_OFF/TURN_ON depending on service calls implemented
         # This should be removed in 2025.1.
-        if not self.supported_features & ClimateEntityFeature.TURN_OFF:
-            if (
-                type(self).async_turn_off is not ClimateEntity.async_turn_off
-                or type(self).turn_off is not ClimateEntity.turn_off
-            ):
-                # turn_off implicitly supported by implementing turn_off method
-                _report_turn_on_off("TURN_OFF", "turn_off")
-                self.__mod_supported_features |= (  # pylint: disable=unused-private-member
-                    ClimateEntityFeature.TURN_OFF
-                )
-            elif self.hvac_modes and HVACMode.OFF in self.hvac_modes:
-                # turn_off implicitly supported by including HVACMode.OFF
-                _report_turn_on_off("off", "turn_off")
-                self.__mod_supported_features |= (  # pylint: disable=unused-private-member
-                    ClimateEntityFeature.TURN_OFF
-                )
+        if self._enable_turn_on_off_backwards_compatibility is False:
+            # Return if integrations have turned off backwards compatibility
```

Comment:
Maybe something like this is clearer
```suggestion
            # Return if integration is migrated
```